File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-11-01/examples-java/virtualMachineImageExamples/VirtualMachineImages_ListWithProperties_MaximumSet_Gen.java

/**
 * Samples for VirtualMachineImages ListWithProperties.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-11-01/examples/
     * virtualMachineImageExamples/VirtualMachineImages_ListWithProperties_MaximumSet_Gen.json
     */
    /**
     * Sample code: VirtualMachineImages_ListWithProperties_MaximumSet.
     * 
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void
        virtualMachineImagesListWithPropertiesMaximumSet(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.virtualMachines().manager().serviceClient().getVirtualMachineImages().listWithPropertiesWithResponse(
            "eastus", "MicrosoftWindowsServer", "WindowsServer", "2022-datacenter-azure-edition", "Properties", 4, "aa",
            com.azure.core.util.Context.NONE);
    }
}

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-11-01/examples-js/virtualMachineImageExamples/VirtualMachineImages_ListWithProperties_MaximumSet_Gen.js
const { ComputeManagementClient } = require("@azure/arm-compute");
const { DefaultAzureCredential } = require("@azure/identity");
require("dotenv/config");

/**
 * This sample demonstrates how to
 *
 * @summary
 * x-ms-original-file: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-11-01/examples/virtualMachineImageExamples/VirtualMachineImages_ListWithProperties_MaximumSet_Gen.json
 */
async function virtualMachineImagesListWithPropertiesMaximumSet() {
  const subscriptionId = process.env["COMPUTE_SUBSCRIPTION_ID"] || "{subscription-id}";
  const location = "eastus";
  const publisherName = "MicrosoftWindowsServer";
  const offer = "WindowsServer";
  const skus = "2022-datacenter-azure-edition";
  const expand = "Properties";
  const top = 4;
  const orderby = "aa";
  const options = {
    top,
    orderby,
  };
  const credential = new DefaultAzureCredential();
  const client = new ComputeManagementClient(credential, subscriptionId);
  const result = await client.virtualMachineImages.listWithProperties(
    location,
    publisherName,
    offer,
    skus,
    expand,
    options,
  );
  console.log(result);
}
